### DISCUSSION TITLE
Too many arguments for 'public constructor(): androidx/biometric/BiometricPrompt.AuthenticationCallback+Cannot create an instance of an abstract class.

### DISCUSSION THREAD

**Discussion by TaskeenAshraf** [skiptools/skip#290] [Q&A]
So i am using this biometric code that works on android side but it is giving me strange errors in SKIP project. I have overriden all methods from  BiometricPrompt.AuthenticationCallback() but still getting issues

```
 biometricPrompt = BiometricPrompt(
                        activity!!, executor,
                        object : BiometricPrompt.AuthenticationCallback() {
                 override func onAuthenticationError(errorCode: Int, errString: CharSequence) {
                        super.onAuthenticationError(errorCode, errString)
                                    }
                 override func onAuthenticationFailed() {
                        super.onAuthenticationFailed()
                         }
                  override func onAuthenticationSucceeded(result: BiometricPrompt.AuthenticationResult) {
                        super.onAuthenticationSucceeded(result)
                        var type = result.authenticationType
                                    }
                                }
                )
```

**Comment by marcprux** [comment]
I don't see that `BiomatricPrompt` constructor at https://developer.android.com/reference/android/hardware/biometrics/BiometricPrompt. It appears that the API needs it to be created with a [BiometricPrompt.Builder](https://developer.android.com/reference/android/hardware/biometrics/BiometricPrompt.Builder), and then you pass the AuthenticationCallback to the [authenticate](https://developer.android.com/reference/android/hardware/biometrics/BiometricPrompt#authenticate(android.os.CancellationSignal,%20java.util.concurrent.Executor,%20android.hardware.biometrics.BiometricPrompt.AuthenticationCallback)) function. Take a look at the example at https://developer.android.com/identity/sign-in/biometric-auth
